gdax/public_client.py

```diff
@@ -161,7 +161,7 @@ def get_product_trades(self, product_id, before='', after='', limit='', result=[
 
         result.extend(r.json())
 
-        if 'cb-after' in r.headers and limit is not len(result):
+        if 'cb-after' in r.headers and limit != len(result):
```

Comment:
What if len(result) > limit. This logic could be simplified so we dont have to do: `limit -= len(result)` in the next line right?